### PR TITLE
Get sensor readings v2/readings

### PR DESCRIPTION
**Pull request by karimkawambwa**
## Description

Under version 2, the `/readings` route will be used to retrieve sensor data.
This PR introduces the routes:

- `/v2/air/readings/<city slug>`
```
[
   {
     "day": "2019-01-30T00:00:00.0Z",
   "value_type" : "P2"
    "average" : 0.0, 
    "minimum" :  0.0, 
    "maximum" :  0.0,
   }, 
  .
  .
  . ,
]
```

- `/v2/air/readings/<city slug>?from=<date>&to=<date>`
```
[
    {
     "day": "2019-01-30T00:00:00.0Z",
     "value_type": "P2",
      "average" : 0.0, 
      "minimum" :  0.0, 
      "maximum" :  0.0,
    },
.
.
.
]
```

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Comment by kilemensi** [comment]
Guess this needs updating @karimkawambwa

**Comment by karimkawambwa** [comment]
Example: http://sensorsafrica-staging.dokku-2.sensors.africa/v2/air/readings/?created__date__range=01/28/2019,02/05/2019&city=Nairobi

@kilemensi

File: sensorsafrica/tests/conftest.py [hunk 1]
```diff
@@ -20,48 +23,147 @@ def django_db_setup(django_db_setup, django_db_blocker):
...
-    )
+    s, x = Sensor.objects.get_or_create(node=node, sensor_type=sensor_type)
     return s
+
+
+@pytest.fixture
+def locations():
+    return SensorLocation.objects.bulk_create(
+        [
+            SensorLocation(city="Dar es Salaam", description="active"),
```
Comment (kilemensi) [inline sensorsafrica/tests/conftest.py]: I'm not too sure about names with spaces... I think `dar-es-salaam` should be the value we store in DB and use to query. Dar es Salaam, DaresSalaam, Dar ès Salaam, etc shouldn't make it to our DB.

Comment (karimkawambwa) [reply sensorsafrica/tests/conftest.py]: This is a valid point, we would have to change all locations in the production database. I will will have to consider this for a story.

Additionally, I don't see any implementation of a route to create a sensor with location. I have to confirm if the deployed sensors were added manually.

**Comment by karimkawambwa** [comment]
> 1. Lets lose the whole `/readings` part of the url. The fact that we're using `/air`, `/water`, etc., already says we're getting air readings.

I think we should be explicit about the route and say /data or /readings.

Moreover GET `/air/readings` would support the formats like GET `/air/sensors`. Additionally our implementations to use PUT/POST/DELETE `/air/readings`, PUT/POST `/air/sensors`...

@kilemensi

**Comment by kilemensi** [comment]
> > 1. Lets lose the whole `/readings` part of the url. The fact that we're using `/air`, `/water`, etc., already says we're getting air readings.
> 
> I think we should be explicit about the route and say /data or /readings.
> 
> Moreover GET `/air/readings` would support the formats like GET `/air/sensors`. Additionally our implementations to use PUT/POST/DELETE `/air/readings`, PUT/POST `/air/sensors`...
> 
> @kilemensi

Why does `/sensors` need to be under `/air`? What if we have a sensor that can measure both air and sound? Sensors needs their own url.

File: sensorsafrica/tests/conftest.py (hunk 1 above)
Comment (kilemensi) [reply sensorsafrica/tests/conftest.py]: Create a story for this and resolve this comment @karimkawambwa

**Comment by kilemensi** [comment]
> > > 1. Lets lose the whole `/readings` part of the url. The fact that we're using `/air`, `/water`, etc., already says we're getting air readings.
> > 
> > 
> > I think we should be explicit about the route and say /data or /readings.
> > Moreover GET `/air/readings` would support the formats like GET `/air/sensors`. Additionally our implementations to use PUT/POST/DELETE `/air/readings`, PUT/POST `/air/sensors`...
> > @kilemensi
> 
> Why does `/sensors` need to be under `/air`? What if we have a sensor that can measure both air quality and sound? Sensors need their own URL.

Having considered everything @karimkawambwa, what have you decided?

**Comment by karimkawambwa** [comment]
> Having considered everything @karimkawambwa, what have you decided?

@kilemensi I think this is a better route naming. This way we access air sensors through `air/sensors` and example water sensors through `water/sensors`.

> What if we have a sensor that can measure both air quality and sound?

At this point we will deal with sensors and not nodes. I think most sensors handle one task. A node is mixed with different sensors. I think if we query sensors, I will return sensors and not the node that is the microcontroller.

**Comment by kilemensi** [comment]
Alright @karimkawambwa. If you've considered all the possibilities and are happy with this direction then I fully support your decision and will proceed to review the PR.

-----------------------------------------------------------------------------------------

 1. For the foreseeable future, data will still be collected using feistaub/v1 routes i.e. `/v1/push-sensor-data/` since we have close to 100 sensor nodes already deployed in the field and I don't think the nodes support over the air reconfiguration. This means there is no need for `/v2` to support data entry yet.
 2. Unlike `/v1/now`, `/air`, `/water`, etc. shall be returning processed/summarized/interpreted values and not the actual sensor readings/measurements. I'm talking averages over a given period, location, etc. So note this when you refer to these as _readings_.
 3. Personally I wouldn't categorize sensors into `air`, `water`, etc. If I had to, I would categorise the _sensor nodes/kits_ ie. such and such is our AQ sensor kit. Categorizing sensors themselves is very challenging/open to interpretation. See this list of sensors on [Wikipedia](https://en.wikipedia.org/wiki/List_of_sensors) for example.

File: sensorsafrica/urls.py [hunk 2]
```diff
@@ -22,13 +22,15 @@
...
 from .accounts.base.router import api_urls as accounts_api_v1
+from .data.base.router import api_urls as data_api_v2
 from .sensors.base.router import api_urls as sensors_api_v1
 
 urlpatterns = [
     url(r"^$", RedirectView.as_view(url="/docs/", permanent=False)),
     url(r"^admin/", admin.site.urls),
     url(r"^v1/", include(sensors_api_v1)),
     url(r"^v1/", include(accounts_api_v1)),
+    url(r"^v2/(?P<sensor_type>[air]+)/readings/", include(data_api_v2)),
```
Comment (kilemensi) [inline sensorsafrica/urls.py]: Why not move `(?P<sensor_type>[air]+)/readings/` into `data_api_v2` so we can simplify this pattern here to just
```python
...
url(r"^v2/", include(data_api_v2)),
...
```
Like how we're doing for `v1`

**Pull request merged**